ingestion/src/metadata/utils/protobuf_parser.py

```diff
@@ -0,0 +1,204 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Utils module to parse the protobuf schema
+"""
+
+import glob
+import importlib
+import os
+import pathlib
+import shutil
+import sys
+import traceback
+from enum import Enum
+from typing import Optional
+
+import grpc_tools.protoc
+from pydantic import BaseModel
+
+from metadata.utils.helpers import snake_to_camel
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class ProtobufDataTypes(Enum):
+    """
+    Enum for Protobuf Datatypes
+    """
+
+    # Field type unknown.
+    TYPE_UNKNOWN = 0
+    # Field type double.
+    TYPE_DOUBLE = 1
+    # Field type float.
+    TYPE_FLOAT = 2
+    # Field type int64.
+    TYPE_INT64 = 3
+    # Field type uint64.
+    TYPE_UINT64 = 4
+    # Field type int32.
+    TYPE_INT32 = 5
+    # Field type fixed64.
+    TYPE_FIXED64 = 6
+    # Field type fixed32.
+    TYPE_FIXED32 = 7
+    # Field type bool.
+    TYPE_BOOL = 8
+    # Field type string.
+    TYPE_STRING = 9
+    # Field type group. Proto2 syntax only, and deprecated.
+    TYPE_GROUP = 10
+    # Field type message.
+    TYPE_MESSAGE = 11
+    # Field type bytes.
+    TYPE_BYTES = 12
+    # Field type uint32.
+    TYPE_UINT32 = 13
+    # Field type enum.
+    TYPE_ENUM = 14
+    # Field type sfixed32.
+    TYPE_SFIXED32 = 15
+    # Field type sfixed64.
+    TYPE_SFIXED64 = 16
+    # Field type sint32.
+    TYPE_SINT32 = 17
+    # Field type sint64.
+    TYPE_SINT64 = 18
+
+
+class ProtobufParserConfig(BaseModel):
+    """
+    Protobuf Parser Config class
+    """
+
+    schema_name: str
+    schema_text: str
+    base_file_path: Optional[str] = "/tmp/protobuf_openmetadata"
```

Comment:
let's document why this is needed and how this is used. It can be here itself in the docstring

Reply:
otherwise magic paths can be tricky

Reply:
Yes I'll add in the docstring